### PR TITLE
Add config for creating case when payments are pending

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/BPS-844

Follow up to https://github.com/hmcts/bulk-scan-orchestrator/pull/724

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java [hunk 1]
```diff
@@ -165,11 +168,10 @@ private ProcessResult tryCreateNewCase(
...
 
-        if (awaitsPaymentProcessing && !ignoreWarnings) {
-            return new ProcessResult(
-                singletonList("Payments for this Exception Record have not been processed yet"),
-                emptyList()
-            );
+        if (awaitsPaymentProcessing && configItem.allowCreatingCaseBeforePaymentsAreProcessed() && !ignoreWarnings) {
+            return new ProcessResult(singletonList(AWAITING_PAYMENTS_MESSAGE), emptyList());
+        } else if (awaitsPaymentProcessing && !configItem.allowCreatingCaseBeforePaymentsAreProcessed()) {
+            return new ProcessResult(emptyList(), singletonList(AWAITING_PAYMENTS_MESSAGE));
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: Could we do something like below ?
```
boolean isCaseCreationAllowed = configItem.allowCreatingCaseBeforePaymentsAreProcessed();
             
             if(awaitsPaymentProcessing){
				return isCaseCreationAllowed==true ?
				new ProcessResult(singletonList(AWAITING_PAYMENTS_MESSAGE), emptyList());
				: new ProcessResult(emptyList(), singletonList(AWAITING_PAYMENTS_MESSAGE));
             }

```

File: src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java [hunk 2]
```diff
@@ -650,10 +653,67 @@ void should_return_error_if_publishing_payment_command_failed() throws Exception
         assertThat(exception.getCause()).isInstanceOf(PaymentsPublishingException.class);
     }
 
+    @ParameterizedTest(name = "Allowed to proceed: {0}. User ignores warnings: {1}")
+    @CsvSource({
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java]: good to know 👍

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java (hunk 1 above)
Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: @nitinprabhu What you suggest excludes ignoreWarnings from the logic

Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: boolean isCaseCreationAllowed = 
 configItem.allowCreatingCaseBeforePaymentsAreProcessed();
              if (awaitsPaymentProcessing) {
                 if (!isCaseCreationAllowed) {
                     return new ProcessResult(emptyList(), singletonList(AWAITING_PAYMENTS_MESSAGE));
                 }
                 if (!ignoreWarnings) {
                     return new ProcessResult(singletonList(AWAITING_PAYMENTS_MESSAGE), emptyList());
                 }

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java]: I **really** want to keep `if-else` style, no hidden exits from the method

**Pull request merged**